vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -223,6 +329,29 @@ public void updateImageBlock(IImageBlock imageBlock, byte[] image, String filena
         imageBlockRepo.save((ImageBlock) imageBlock);
     }
 
+    @Override
+    public void updateVideoBlock(IVideoBlock videoBlock, byte[] video, Long fileSize, String url, String filename, Integer contentOrder)
+            throws VideoCouldNotBeStoredException {
+
+        if(video != null ) {
+            IVSVideo slideContentVideo = saveVideo(video, fileSize, filename);
+            storeVideoFile(video, slideContentVideo, filename);
+            slideContentVideo.setUrl(null);
+            videoBlock.setVideo(slideContentVideo);
+        }
+        else {
+            
+            videoBlock.getVideo().setUrl(url);
+            videoBlock.getVideo().setFilename(null);
+            videoBlock.getVideo().setFileType(null);
+            videoBlock.getVideo().setFileSize(null);
+            videoBlock.getVideo().setHeight(0);
+            videoBlock.getVideo().setWidth(0);
+            videoBlock.getVideo().setParentPath(null);
```

Comment:
the video still exists, doesn't it? It's not deleted from the file system? If so, the vide object should still exist so it can be found. Just the references from the block to the video should be deleted.